subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CachedTaskExecutionIntegrationTest.groovy

```diff
@@ -305,6 +305,22 @@ class CachedTaskExecutionIntegrationTest extends AbstractIntegrationSpec impleme
         file("build/classes/main/Hello.class").exists()
     }
 
+    def "Using `doNotCacheIf` without reason is deprecated"() {
+        given:
+        buildFile << """
+            task adHocTask {
+                outputs.doNotCacheIf { true }
+            }
+        """
+
+        when:
+        executer.expectDeprecationWarning()
+        withBuildCache().succeeds 'adHocTask'
+
+        then:
+        output.contains "The doNotCacheIf(Spec) method has been deprecated and is scheduled to be removed in Gradle 4.0. Please use the doNotCacheIf(String, Spec) method instead."
+    }
+
```

Comment:
I was thinking not to deprecate `cacheIf` without a reason since it can be used to turn caching on:
```groovy
cacheIf {
   true
}
```
If we required a reason in that case (could be as easy as "Turn caching on"), then the reason would currently not be printed - so I was thinking why should we force that on the user. What is your take on that?